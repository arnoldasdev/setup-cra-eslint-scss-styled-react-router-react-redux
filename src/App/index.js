import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router } from 'react-router-dom';
import { connect } from 'react-redux';
import Layout from 'components/Layout';
import { Header, Footer } from 'components/views';
import { Authenticated, Login } from 'components/switches';
import './index.scss';

const App = ({ auth }) => (
  <div className="App" data-testid="App">
    <Router>
      {auth.isAuthenticated
        ? <Layout views={{ Content: Authenticated, Header, Footer }} />
        : <Layout views={{ Content: Login, Footer }} />}
    </Router>
  </div>
);

App.propTypes = {
  auth: PropTypes.exact({
    isAuthenticated: PropTypes.bool.isRequired,
  }).isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps)(App);
