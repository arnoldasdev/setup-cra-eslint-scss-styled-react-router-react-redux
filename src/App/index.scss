@use 'styles';
@use 'styles/scss/var/colors';

.App {
  color: colors.$light;
  background-color: colors.$dark;
  min-height: 100vh;
  font-size: calc(1rem + 0.25vw);
  word-break: break-word;
}
