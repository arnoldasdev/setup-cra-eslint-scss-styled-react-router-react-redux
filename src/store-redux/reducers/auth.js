const initial = { isAuthenticated: false };

export const auth = (state = initial, action) => {
  switch (action.type) {
    case 'LOG_IN':
      return { isAuthenticated: true };
    case 'LOG_OUT':
      return { isAuthenticated: false };
    default:
      return state;
  }
};
