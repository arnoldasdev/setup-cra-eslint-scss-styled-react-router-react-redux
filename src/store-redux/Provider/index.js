import React from 'react';
import PropTypes from 'prop-types';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducers from 'store-redux/reducers';

/* eslint-disable no-underscore-dangle */

export const store = createStore(
  reducers,
  process.env.NODE_ENV === 'development'
    ? window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    : undefined,
);

const Container = ({ children }) => (
  <Provider store={store}>
    {children}
  </Provider>
);

Container.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Container;
