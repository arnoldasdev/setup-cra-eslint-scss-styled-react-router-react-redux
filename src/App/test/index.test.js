import React from 'react';
import { render, cleanup, act } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Provider, { store } from 'store-redux/Provider';
import { login, logout } from 'store-redux/actions/auth';
import App from '..';

afterEach(cleanup);

describe('[main]', () => {
  describe('User-is-Authenticated', () => {
    it('<App /> renders-correctly', () => {
      act(() => {
        store.dispatch(login());
      });

      const { getByTestId } = render(
        <Provider>
          <App />
        </Provider>,
      );

      const app = getByTestId('App');
      expect(app).toBeInstanceOf(HTMLDivElement);
    });

    it('<Layout /> is-in-the <App />', () => {
      act(() => {
        store.dispatch(login());
      });

      const { getByTestId } = render(
        <Provider>
          <App />
        </Provider>,
      );

      const app = getByTestId('App');
      const layout = getByTestId('Layout');
      expect(app.firstChild).toEqual(layout);
    });

    it('<Header />, <Content />, <Footer /> is-in-the <Layout />', () => {
      act(() => {
        store.dispatch(login());
      });

      const { getByTestId } = render(
        <Provider>
          <App />
        </Provider>,
      );

      const layout = getByTestId('Layout');
      const header = getByTestId('Header');
      const content = getByTestId('Content');
      const footer = getByTestId('Footer');

      [header, content, footer].forEach((child, i) => expect(child).toEqual(layout.children[i]));
    });
  });

  describe('User-is-not-Authenticated', () => {
    it('<App /> renders-correctly', () => {
      act(() => {
        store.dispatch(logout());
      });

      const { getByTestId } = render(
        <Provider>
          <App />
        </Provider>,
      );

      const app = getByTestId('App');
      expect(app).toBeInstanceOf(HTMLDivElement);
    });

    it('<Layout /> is-in-the <App />', () => {
      act(() => {
        store.dispatch(logout());
      });

      const { getByTestId } = render(
        <Provider>
          <App />
        </Provider>,
      );

      const app = getByTestId('App');
      const layout = getByTestId('Layout');
      expect(app.firstChild).toEqual(layout);
    });

    it('<Content />, <Footer /> is-in-the <Layout />', () => {
      act(() => {
        store.dispatch(logout());
      });

      const { getByTestId } = render(
        <Provider>
          <App />
        </Provider>,
      );

      const layout = getByTestId('Layout');
      const content = getByTestId('Content');
      const footer = getByTestId('Footer');

      [content, footer].forEach((child, i) => expect(child).toEqual(layout.children[i]));
    });
  });
});

describe('[snapshots]', () => {
  describe('User-is-Authenticated', () => {
    it('Matches-snapshot <App>', () => {
      renderer.act(() => {
        store.dispatch(login());
      });

      const tree = renderer
        .create(
          <Provider>
            <App />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('User-is-not-Authenticated', () => {
    it('Matches-snapshot <App>', () => {
      renderer.act(() => {
        store.dispatch(logout());
      });

      const tree = renderer
        .create(
          <Provider>
            <App />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
