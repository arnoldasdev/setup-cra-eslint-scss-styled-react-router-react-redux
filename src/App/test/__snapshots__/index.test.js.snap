// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[snapshots] User-is-Authenticated Matches-snapshot <App> 1`] = `
<div
  className="App"
  data-testid="App"
>
  <div
    className="Layout"
    data-testid="Layout"
  >
    <header
      className="Header"
      data-testid="Header"
    >
      <div
        className="view-Header"
      >
        <div
          className="view-Header-logo"
        >
          <img
            alt="logo"
            src="logo.svg"
          />
        </div>
        <nav
          className="view-Header-nav"
        >
          <a
            href="/"
            title="/"
          >
            /
          </a>
          <a
            href="/"
            onClick={[Function]}
            title="home"
          >
            /home
          </a>
          <a
            href="/fetch"
            onClick={[Function]}
            title="fetch"
          >
            /fetch
          </a>
          <a
            href="/about"
            onClick={[Function]}
            title="about"
          >
            /about
          </a>
          <a
            href="/redirect"
            onClick={[Function]}
            title="redirect to /"
          >
            /redirect
          </a>
          <button
            className="Button"
            onClick={[Function]}
            title="Log out"
            type="button"
          >
            Log out
          </button>
        </nav>
      </div>
    </header>
    <div
      className="Content"
      data-testid="Content"
    >
      <div
        className="view-Home"
      >
        <h2>
          [setup]-create-react-app
        </h2>
        <ul>
          <li>
            react-redux
          </li>
          <li>
            react-router
          </li>
          <li>
            styled-components
          </li>
          <li>
            scss
          </li>
          <li>
            eslint [airbnb]
          </li>
        </ul>
        <div
          className="Samples"
        >
          <div
            className="Row Row-1-col ta-center"
          >
            <div
              className="col"
            >
              <h5>
                redux-state
              </h5>
              <button
                className="Button"
                onClick={[Function]}
                title="Log out"
                type="button"
              >
                Log out
              </button>
            </div>
          </div>
          <div
            className="Row Row-2-col"
          >
            <div
              className="col"
            >
              <div
                className="Box"
              >
                <div
                  className="Box-background"
                />
                <div
                  className="Box-body"
                >
                  <span
                    className="Box-content"
                  >
                    Sample: SCSS-component
                  </span>
                  <a
                    className="Anchor AnchorButton Box-button"
                    href="https://sass-lang.com/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Learn More
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="sc-dlnjPT bkhSxZ"
              >
                <div
                  className="sc-hKFyIo iuyJch"
                />
                <div
                  className="sc-eCApGN hzSwdj"
                >
                  <span
                    className="sc-jSFkmK eHhjwz"
                  >
                    Sample: styled-component
                  </span>
                  <a
                    className="sc-bdnylx sc-gtssRu hCXnCn ljNLJV sc-gKAblj laHkcQ"
                    href="https://styled-components.com/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Learn More
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      className="Footer"
      data-testid="Footer"
    >
      <div
        className="view-Footer"
      >
        <div
          className="view-Footer-react-default-content"
        >
          <p>
            Edit src/components and save to reload.
          </p>
          <a
            className="Anchor view-Footer-anchor"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn React
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;

exports[`[snapshots] User-is-not-Authenticated Matches-snapshot <App> 1`] = `
<div
  className="App"
  data-testid="App"
>
  <div
    className="Layout"
    data-testid="Layout"
  >
    <div
      className="Content"
      data-testid="Content"
    >
      <div
        className="view-Login"
      >
        <div
          className="Row Row-1-col ta-center"
        >
          <div
            className="col"
          >
            <div
              className="ta-center"
            >
              <h2>
                /login
              </h2>
            </div>
          </div>
          <div
            className="col"
          >
            <h5>
              redux-state
            </h5>
            <button
              className="Button"
              onClick={[Function]}
              title="Log in"
              type="button"
            >
              Log in
            </button>
          </div>
        </div>
      </div>
    </div>
    <footer
      className="Footer"
      data-testid="Footer"
    >
      <div
        className="view-Footer"
      >
        <div
          className="view-Footer-react-default-content"
        >
          <p>
            Edit src/components and save to reload.
          </p>
          <a
            className="Anchor view-Footer-anchor"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn React
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
